receiver/nginxreceiver/scraper.go

```diff
@@ -87,6 +85,6 @@ func (r *nginxScraper) scrape(context.Context) (pdata.Metrics, error) {
 	r.mb.RecordNginxConnectionsCurrentDataPoint(now, stats.Connections.Writing, metadata.AttributeState.Writing)
 	r.mb.RecordNginxConnectionsCurrentDataPoint(now, stats.Connections.Waiting, metadata.AttributeState.Waiting)
 
-	r.mb.Emit(ilm.Metrics())
+	r.mb.Emit(md.ResourceMetrics().At(0).InstrumentationLibraryMetrics().At(0).Metrics())
```

Comment:
This doesn't necessarily need to be part of this PR, but taking this a step further:

Since the metric builder is initializing the `InstrumentationLibraryMetrics`, it seems that it should not be required to drill down to retrieve the same. Perhaps instead, the `Emit` function should assume that it is emitting the `InstrumentationLibraryMetrics` that it initialized?

Reply:
I agree here. We still probably need to provide a way to customize the `pdata.Metric` object, but it can be done after calling `Emit`.

I believe we can avoid exposing `NewMetricData` and handle it internally by the first call to `Record...DataPoint`. And `Emit` signature can be changed to:

`func (mb *MetricsBuilder) Emit() pdata.Metric`

@djaglowski @codeboten WDYT?

Reply:
It might be more involved than it seems. So we can also merge this PR as is and apply the suggested improvement after that, it can be added as additional improvement in [this issue](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/6429)

Reply:
@dmitryax, do I get it right that you approved this PR as it is then?

Reply:
Yes, I can approve it as is. Just wanted to hear back from @codeboten

Reply:
Created an issue https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/8358 for this

Reply:
Thanks!